spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/util/FormUtils.java

```diff
@@ -1232,43 +1207,52 @@ public static String getPlatePosition(int rowCount) {
     return samples;
   }
 
-  public static void createBoxContentsSpreadsheet(File outpath, ArrayList<String> array) throws IOException {
-    InputStream in = null;
-    in = FormUtils.class.getResourceAsStream("/forms/ods/box_input.xlsx");
+  public static void createPlainBoxSpreadsheet(File outpath, String name, String alias, List<List<String>> boxContents)
+      throws IOException {
+    createBoxSpreadsheet("/forms/ods/box_input_plain.xlsx", outpath, name, alias, boxContents);
+  }
+
+  public static void createDetailedBoxSpreadsheet(File outpath, String name, String alias, List<List<String>> boxContents)
+      throws IOException {
+    createBoxSpreadsheet("/forms/ods/box_input_detailed.xlsx", outpath, name, alias, boxContents);
+  }
+
+  private static void createBoxSpreadsheet(String templateName, File outpath, String name, String alias, List<List<String>> boxContents) throws IOException {
+    InputStream in = FormUtils.class.getResourceAsStream(templateName);
+
     if (in != null) {
       XSSFWorkbook oDoc = new XSSFWorkbook(in);
 
       XSSFSheet sheet = oDoc.getSheet("Input");
       FileOutputStream fileOut = new FileOutputStream(outpath);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "FileOutputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)

Reply:
👍 try-with-resources